test/e2e/bgppolicy_test.go

```diff
@@ -130,77 +123,193 @@ func TestBGPPolicy(t *testing.T) {
+		asnStart := int32(61000)
+
+		t.Log("Configure the remote FRR router with BGP")
+		configureExternalBGPRouter(t, externalASN, confederationIdentifier, true)
+
+		var bgpPolicies []*crdv1alpha1.BGPPolicy
+		for i := 0; i < len(clusterInfo.nodes); i++ {
+			bgpPolicyName := "test-policy-" + strconv.Itoa(i)
+			bgpPolicy := &crdv1alpha1.BGPPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: bgpPolicyName,
+				},
+				Spec: crdv1alpha1.BGPPolicySpec{
+					NodeSelector: metav1.LabelSelector{
+						MatchLabels: map[string]string{},
+					},
+					LocalASN:   asnStart + int32(i),
+					ListenPort: ptr.To[int32](179),
+					Confederation: &crdv1alpha1.Confederation{
+						Identifier: confederationIdentifier,
+					},
+					Advertisements: crdv1alpha1.Advertisements{
+						Service: &crdv1alpha1.ServiceAdvertisement{
+							IPTypes: []crdv1alpha1.ServiceIPType{crdv1alpha1.ServiceIPTypeClusterIP},
+						},
+						Pod: &crdv1alpha1.PodAdvertisement{},
+					},
+					BGPPeers: []crdv1alpha1.BGPPeer{
+						{Address: externalInfo.externalFRRIPv4, ASN: externalASN},
+					},
+				},
+			}
+			for j := 0; j < len(clusterInfo.nodes); j++ {
+				if j == i {
+					continue
+				}
+				asn := asnStart + int32(j)
+				bgpPolicy.Spec.Confederation.MemberASNs = append(bgpPolicy.Spec.Confederation.MemberASNs, asn)
+				bgpPolicy.Spec.BGPPeers = append(bgpPolicy.Spec.BGPPeers, crdv1alpha1.BGPPeer{Address: clusterInfo.nodes[j].ipv4Addr, ASN: asn})
+			}
+			bgpPolicy, err = data.crdClient.CrdV1alpha1().BGPPolicies().Create(context.TODO(), bgpPolicy, metav1.CreateOptions{})
+			defer data.crdClient.CrdV1alpha1().BGPPolicies().Delete(context.TODO(), bgpPolicyName, metav1.DeleteOptions{})
+			require.NoError(t, err)
+			bgpPolicies = append(bgpPolicies, bgpPolicy)
+		}
 
-	t.Log("Get routes installed on remote FRR router and verify them")
-	expectedRoutes = []FRRRoute{{Prefix: clusterIP + "/32", Nexthops: []string{nodeIPv4(0)}}}
-	notExpectedRoutes := []FRRRoute{{Prefix: clusterIP + "/32", Nexthops: getAllNodeIPs()}}
-	checkFRRRouterBGPRoutes(t, expectedRoutes, notExpectedRoutes)
+		t.Log("Get the routes installed on remote FRR router and verify them")
+		expectedRoutes := make([]FRRRoute, 0)
+		for _, node := range clusterInfo.nodes {
+			expectedRoutes = append(expectedRoutes, FRRRoute{Prefix: node.podV4NetworkCIDR, Nexthops: []string{node.ipv4Addr}})
+		}
+		expectedRoutes = append(expectedRoutes, FRRRoute{Prefix: clusterIP + "/32", Nexthops: getAllNodeIPs()})
+		checkFRRRouterBGPRoutes(t, expectedRoutes, nil)
+
+		t.Log("Verify the connectivity of the installed routes on remote FRR route")
+		ipsToConnect := []string{podIP, clusterIP}
+		for _, ip := range ipsToConnect {
+			cmd := fmt.Sprintf("/usr/bin/wget -O - http://%s:8080/hostname -T 5", ip)
+			rc, stdout, _, err := exec.RunDockerExecCommand(externalInfo.externalFRRCID, cmd, "/", nil, "")
+			require.NoError(t, err)
+			require.Equal(t, 0, rc)
+			require.Equal(t, podName, stdout)
+		}
```

Comment:
I think this is repeated multiple times, could you define a helper function - ideally inside the test function

Reply:
Done